packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,202 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def _name_heuristic(argument, field_name, used_col_names):
+    if isinstance(argument, int):
+        argument = str(argument)
+    if field_name not in used_col_names:
+        return field_name
+    elif field_name + argument not in used_col_names:
+        return field_name + "_" + argument
+    else:
+        raise NameError(
+            "A name conflict was encountered for argument %s."
+            "Columns with names %s, %s and %s are already used"
+            % (field_name, argument, field_name, field_name + "_" + argument)
+        )
+
+
+def _initialize_argument_col_names(args, attrables, array_attrables):
+    """
+    This function builds a list of columns of the data_frame argument used
+    as arguments, either as str/int arguments or given as columns
+    (pandas series type).
+    """
+    df = args["data_frame"]
+    used_col_names = set()
+    for field in args:
+        if field not in attrables:
+            continue
+        names = args[field] if field in array_attrables else [args[field]]
+        if names is None:
+            continue
+        for arg in names:
+            if arg is None:
+                continue
+            if isinstance(arg, str):
+                used_col_names.add(arg)
+            if isinstance(arg, pd.DataFrame) or isinstance(arg, pd.core.series.Series):
```

Comment:
we accept full `DataFrame`s as well as `Series`?

Reply:
hum now, it won't work with a full `DataFrame` actually (an error will be raised because it has no `name`). I will correct the code and the doc.

Reply:
ok, corrected